symfony/plugins/orangehrmLeavePlugin/Api/LeaveAPI.php

```diff
@@ -0,0 +1,230 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Leave\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Exception\ForbiddenException;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\Service\NormalizerServiceTrait;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\LeaveRequest;
+use OrangeHRM\Leave\Api\Model\LeaveDetailedModel;
+use OrangeHRM\Leave\Api\Traits\LeaveRequestParamHelperTrait;
+use OrangeHRM\Leave\Dto\LeaveSearchFilterParams;
+use OrangeHRM\Leave\Service\LeaveRequestCommentService;
+use OrangeHRM\Leave\Traits\Service\LeaveRequestServiceTrait;
+use OrangeHRM\Pim\Api\Model\EmployeeModel;
+
+class LeaveAPI extends Endpoint implements CrudEndpoint
+{
+    use LeaveRequestParamHelperTrait;
+    use LeaveRequestServiceTrait;
+    use UserRoleManagerTrait;
+    use NormalizerServiceTrait;
+
+    public const FILTER_LEAVE_REQUEST_ID = 'leaveRequestId';
+    public const META_PARAMETER_EMPLOYEE = 'employee';
+
+    /**
+     * @var null|LeaveRequestCommentService
+     */
+    protected ?LeaveRequestCommentService $leaveRequestCommentService = null;
+
+    /**
+     * @return LeaveRequestCommentService
+     */
+    public function getLeaveRequestCommentService(): LeaveRequestCommentService
```

Comment:
wrong usage. We should move Dao functionalities and tests to LeaveRequestDao, LeaveRequestDaoTest